__tests__/unit/integration/api.test.js

```diff
@@ -0,0 +1,148 @@
+  });
+
+  // tests for route "/:num([-0-9.,]+)"
+
+  test("batch request with only range", async function () {
+    const res = await request(app).get("/1..3");
+    expect(res.body).toHaveProperty("1", "2", "3");
+  });
+
+  test("batch request with range and specific number", async function () {
+    const res = await request(app).get("/1..3,10");
+    expect(res.body).toHaveProperty("1", "2", "3", "10");
+  });
+
+  test("invalid url for batch request", async function () {
+    const res = await request(app).get("/1...3");
+    expect(res.text).toEqual("Invalid url");
+  });
+
+  test("batch request with type 'date'", async function () {
+    const res = await request(app).get("/3..5/date");
+    expect(res.body[3]).toContain("January 3rd");
+  });
+
+  test("batch request with type 'math'", async function () {
+    const res = await request(app).get("/3..5/math");
+    let resWords = res.body[3].split(" ");
+    expect(Number(resWords[0])).toEqual(3);
+  });
+
+  test("batch request with type 'trivia'", async function () {
+    const res = await request(app).get("/3..5/trivia");
+    let resWords = res.body[3].split(" ");
+    expect(Number(resWords[0])).toEqual(3);
+  });
+
+  test("batch request with type 'year'", async function () {
+    const res = await request(app).get("/3..5/year");
+    // has "is the year"
+    expect(res.body[3]).toContain("is the year");
+  });
+
+  // tests for route "/:month(-?[0-9]+)/:day(-?[0-9]+)/:type(date)?"
+
+  test("month/day form with type 'date'", async function () {
+    const res = await request(app).get("/2/3/date");
+    expect(res.text).toContain("February 3rd");
+  });
+
+  test("month/day form without type defined", async function () {
+    const res = await request(app).get("/2/3");
+    expect(res.text).toContain("February 3rd");
+  });
+
+  test("month/day form with month above 12", async function () {
+    const res = await request(app).get("/13/8/date");
+    expect(res.text).toContain("January 8th");
+  });
+
+  test("invalid month/day form", async function () {
+    const res = await request(app).get("/2/3/math");
+    expect(res.statusCode).toBe(404);
+  });
+
+  // tests for route "/:date([-0-9/.,]+)/:type(date)?"
+
+  // tests for route "/random/:type?"
+
+  test("random with without type defined", async function () {
+    const res = await request(app).get("/random");
+    let resWords = res.text.split(" ");
+    expect(typeof Number(resWords[0])).toEqual("number");
+  });
+
+  test("random with type 'date'", async function () {
+    const res = await request(app).get("/random/date");
+    let resWords = res.text.split(" ");
+    let months = [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ];
+    expect(months).toContain(resWords[0]);
+  });
+
+  test("random with type 'math'", async function () {
```

Comment:
yeah same for all of these, i think mocking Math.random (or whatever is used to do the randomness) is the way to go here, since then these api calls will be deterministic.